compiler-core/src/erlang/tests/snapshots/gleam_core__erlang__tests__external_fn__no_target_supported_function_fault_tolerance.snap

```diff
@@ -12,6 +12,16 @@ The `no_impl` function is public but doesn't have an implementation for
 the JavaScript target. All public functions of a package must be able to
 compile for a module to be valid.
 
+error: Unsupported target
+  ┌─ /src/one/two.gleam:6:1
+  │
+6 │ pub fn main() {
+  │ ^^^^^^^^^^^^^
+
+The `main` function is public but doesn't have an implementation for
+the JavaScript target. All public functions of a package must be able to
+compile for a module to be valid.
+
```

Comment:
Hmmm yeah i guess now it reports both the statement that is unsupported and the function itself. I don't think the behavior as implemented is wrong but it would be annoying.

Reply:
Ok investigating this more, it looks like we actually have no cases/tests right now that generate this error. Is this error ever possible to generate on main since when a function fails to infer we say it supports every target https://github.com/gleam-lang/gleam/blob/main/compiler-core/src/analyse.rs#L502 what do you think? should we drop this error entirely? I kinda like including this but I could see it just being redundent

Reply:
I'm confused, how do we have a test for it if it is not reachable?

Reply:
The tests we have that trigger this don't have function bodies and are purely externals https://github.com/search?q=repo%3Agleam-lang%2Fgleam+%22public+but+doesn%27t+have+an+implementation%22&type=code

Reply:
Oh I misread the error/misunderstood this. Lemme see if I can fix this to not double emit